mlflow/xgboost.py

```diff
@@ -152,14 +152,16 @@ def save_model(
 
     # Save an XGBoost model
     xgb_model.save_model(model_data_path)
-
+    xgb_model_class = xgb_model.__class__.__name__
     pyfunc.add_to_model(
         mlflow_model,
         loader_module="mlflow.xgboost",
-        data=model_data_subpath,
```

Comment:
@harupy @jwyyy I think this is a good idea for MLflow 2.0 (whenever we decide to release a 2.0 version). This would be a breaking change for all MLflow Model flavors / loader modules defined outside of the main MLflow code base, as these loader modules would need to be updated to expect a directory rather than a file.

Reply:
I agree :)